cloudinit/config/cc_lxd.py

```diff
@@ -81,6 +160,43 @@
 __doc__ = get_meta_doc(meta)
 
 
+def supplemental_schema_validation(
+    init_cfg: dict, bridge_cfg: dict, preseed_str: str
+):
+    """Validate user-provided lxd network and bridge config option values.
+
+    @raises: ValueError describing invalid values provided.
+    """
+    errors = []
+    if not isinstance(init_cfg, dict):
+        errors.append(
+            f"lxd.init config must be a dictionary. found a"
+            f" '{type(init_cfg).__name__}'",
+        )
+
+    if not isinstance(bridge_cfg, dict):
+        errors.append(
+            f"lxd.bridge config must be a dictionary. found a"
+            f" '{type(bridge_cfg).__name__}'",
+        )
+
+    if not isinstance(preseed_str, str):
+        errors.append(
+            f"lxd.preseed config must be a string. found a"
+            f" '{type(preseed_str).__name__}'",
+        )
+    if preseed_str and (init_cfg or bridge_cfg):
```

Comment:
Yes I think mutual exclusion in our schema is something we need to better support in schema and erroring/warnings. But, I do think we need to deal with this in a separate PR because there are a couple of places where we need to think about how we approach the schema definition and the handling or errors in that regard.

Mutual exclusion cases:
- users:ssh_authorized_keys/ssh_import_id and ssh_redirect_user
- users:snapuser vs name
- lxd:init|network vs lxd:preseed

Reply:
Added https://warthogs.atlassian.net/browse/SC-1288 for tracking